### DISCUSSION TITLE
Error using spring repository during threading

### DISCUSSION THREAD

**Discussion by treyshotz** [quarkusio/quarkus#25249] [Q&A]
Hi, 
I'm having some trouble using Spring Repository when I'm using a thread pool.

I am getting;
`java.lang.RuntimeException: Error injecting io.quarkus.arc.runtime.devconsole.InvocationsMonitor io.quarkus.arc.runtime.devconsole.InvocationInterceptor.invocationMonitor` 

The exception happens in this code
```
  public void saveTable(Match match) throws IllegalArgumentException {
    Context.current()
        .wrap(
            () -> {
              repository.save(convertMatchToModel(match));
            }).run();
  }
 ```
 
 The repository is injected this way:
 ```
   @Inject SdsMatchRepository sdsMatchRepository;
```
The pool is created in this way:
```
    var pool = Executors.newFixedThreadPool(1);
    pool.execute(() -> new updatesThread().start());
```
which eventually leads to the exception above.

I do not get what I am doing wrong. Can anybody help

[a comment by an automated account is not shown]

**Reply by geoand** [reply]
Is this only happening in dev mode?

**Reply by treyshotz** [reply]
Hi @geoand !
I think I actually solved it now! There was an issue with how I was starting the application. I will check just to be sure, but if you don't hear from me again I have probably solved it!

Thanks for the swift reply! I really appreciate it!

**Reply by geoand** [reply]
Great!
